### DISCUSSION TITLE
add aiopenapi3 to software using pydantic

### DISCUSSION THREAD

**Discussion by commonism** [pydantic/pydantic#4451] [Projects using Pydantic]
### Initial Checks

- [X] I have searched Google & GitHub for similar requests and couldn't find anything
- [X] I have read and followed [the docs](https://pydantic-docs.helpmanual.io) and still think this feature is missing

### Description

Hi,

maybe the first API client using pydantic and a great match to interface FastAPI services, I propose to link [aiopenapi3](https://github.com/commonism/aiopenapi3) to the [list of software using pydantic](https://github.com/pydantic/pydantic/compare/master...commonism:pydantic:aiopenapi3).
aiopenapi3 is not a code generator, just feed it the description document and you'll be served.

pydantic is used to parse the description document and create the models defined in description document.
This [even works](https://github.com/commonism/aiopenapi3/blob/9ef36a70daee8a493ad1900ce428e89a357f47e6/tests/model_test.py#L123) for [discriminated models](https://github.com/commonism/aiopenapi3/blob/9ef36a70daee8a493ad1900ce428e89a357f47e6/tests/api/v2/schema.py), you may recognize the [code](https://github.com/pydantic/pydantic/blob/a69136d2096e327ed48954a298c775dceb1551d1/docs/examples/types_union_discriminated_nested.py).

httpx is used for communication, httpx_socks can be used for SOCKS5 DynamicForward via ssh.

aiopenapi3 provides a capable plugin interface which can be used to dr. description documents/requests/responses (similar to dr from suds) in case the a … non conforming dialect …is used and requires … translation.

aiopenapi3 provides an asyncio and std/blocking io interface.

There is limitations from using pydantic, e.g. properties can not be mixed with additionalProperties (due \_\_root\_\_ can not be mixed …), unknown properties have to be ignored by default so they do not overwrite BaseModel object methods, but that's problems resulting from the OpenAPI protocol definition being very liberal, making it quite impossible to implement according to the specification.
But overall the use of pydantic is a great benefit, the validation & error messages are top notch.

### Affected Components

- [ ] [Compatibility between releases](https://pydantic-docs.helpmanual.io/changelog/)
- [ ] [Data validation/parsing](https://pydantic-docs.helpmanual.io/usage/models/#basic-model-usage)
- [ ] [Data serialization](https://pydantic-docs.helpmanual.io/usage/exporting_models/) - `.dict()` and `.json()`
- [ ] [JSON Schema](https://pydantic-docs.helpmanual.io/usage/schema/)
- [ ] [Dataclasses](https://pydantic-docs.helpmanual.io/usage/dataclasses/)
- [ ] [Model Config](https://pydantic-docs.helpmanual.io/usage/model_config/)
- [ ] [Field Types](https://pydantic-docs.helpmanual.io/usage/types/) - adding or changing a particular data type
- [ ] [Function validation decorator](https://pydantic-docs.helpmanual.io/usage/validation_decorator/)
- [ ] [Generic Models](https://pydantic-docs.helpmanual.io/usage/models/#generic-models)
- [ ] [Other Model behaviour](https://pydantic-docs.helpmanual.io/usage/models/) - `construct()`, pickling, private attributes, ORM mode
- [ ] [Settings Management](https://pydantic-docs.helpmanual.io/usage/settings/)
- [X] [Plugins](https://pydantic-docs.helpmanual.io/) and integration with other tools - mypy, FastAPI, python-devtools, Hypothesis, VS Code, PyCharm, etc.

**Comment by samuelcolvin** [comment]
Sorry for the delay, thanks for creating the issue. I'm going to migrate this to a discussion under the right category.

You might also consider adding this to https://github.com/Kludex/awesome-pydantic
